spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/LoadBalancerAutoConfiguration.java

```diff
@@ -65,12 +67,35 @@ public void afterSingletonsInstantiated() {
 		};
 	}
 
+	@Bean
+	public LoadBalancerRequestTransformer uriReconstructingRequestTransformer() {
+		return new LoadBalancerRequestTransformer() {
+
+			@Override
+			public HttpRequest transformRequest(HttpRequest request, ServiceInstance instance,
+					LoadBalancerClient loadBalancer) {
+				return new UriReconstructingRequestWrapper(request, instance, loadBalancer);
```

Comment:
>This default behaviour is provided as a LoadBalancingRequestTransformer so if an application wanted to replace it entirely it could do so.

Another approach would be to hard code this behaviour into LoadBalancerRequestFactory, and if you wanted to replace that behaviour, provide your own implementation of LoadBalancerRequestFactory that overrides the behaviour